src/databricks/labs/remorph/reconcile/recon_capture.py

```diff
@@ -404,6 +469,114 @@ def _insert_into_details_table(
                 recon_table_id, schema_output.compare_df, "schema", schema_output.is_valid
             )
 
+    def _get_df(
+        self,
+        recon_table_id: int,
+        agg_data: DataReconcileOutput,
+        recon_type: str,
+    ):
+
+        column_count = agg_data.mismatch_count
+        agg_df = agg_data.mismatch.mismatch_df
+        match recon_type:
+            case "missing_in_source":
+                column_count = agg_data.missing_in_src_count
+                agg_df = agg_data.missing_in_src
+            case "missing_in_target":
+                column_count = agg_data.missing_in_tgt_count
+                agg_df = agg_data.missing_in_tgt
+
+        if column_count > 0 and agg_df:
+            return self._create_map_column(
+                recon_table_id,
+                agg_df,
+                recon_type,
+                False,
+            )
+        return None
+
+    @classmethod
+    def _union_dataframes(cls, agg_df: DataFrame | None, df: DataFrame) -> DataFrame:
```

Comment:
Not sure why we need this function, it is just result aggregation.

Reply:
Rather than writing the Aggregate metrics for every rule, I am combining (UnionByName) all the Metrics per Table and writing them as a Single DataFrame, which would be a single Transaction in the Delta Table.

Reply:
can't we have a list of df and apply reduce to create a union dataframe.

Reply:
Updated the function